### PR TITLE
Make build more coherent with same "dotnet sdk" on all CI machine.

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
Related to https://github.com/tonerdo/coverlet/pull/390#issuecomment-488669074

We have two script

```
// default is debug
./build.sh debug/release <- unix 
build.cmd debug/release  <- win
```

They download sdk to `reporoot\.tools\` and run dotnet cli from there, this should lead to more coherent build/test result because build is "machine dotnet runtime version agnostic".

/cc @tonerdo @petli

### CODE REVIEW

File: eng/build.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+echo "Downloading '$install_script_url'"
+
+ # Use curl if available, otherwise use wget
+if command -v curl > /dev/null; then
+  curl "$install_script_url" -sSL --retry 10 --create-dirs -o "$install_script"
+else
+  wget -q -O "$install_script" "$install_script_url"
+fi
+
+bash "$install_script" --version 2.2.203 --install-dir "$dotnet_root" || {
```
Comment (MarcoRossignoli) [inline eng/build.sh]: we could move version in a simple text file on eng folder and take version from there.

File: eng/build.ps1 [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+{
+    set-strictmode -version 2.0
+    $ErrorActionPreference = "Stop"
+    [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12  
+
+     $toolsRoot = Join-Path $PSScriptRoot "..\.tools"
+     Create-Directory $toolsRoot
+     $scriptPath = Join-Path $toolsRoot "dotnet-install.ps1"
+     Invoke-WebRequest "https://dot.net/v1/dotnet-install.ps1" -OutFile $scriptPath
+     & $toolsRoot\dotnet-install.ps1 -InstallDir $toolsRoot\dotnet -Version 2.2.203
```
Comment (MarcoRossignoli) [inline eng/build.ps1]: we could move version in a simple text file on eng folder and take version from there.

**Comment by MarcoRossignoli** [comment]
@tonerdo seems work, I moved docs under Documentation in line with `Join foundation` issue.

**Comment by MarcoRossignoli** [comment]
@tonerdo  we should move to AzurePipelines https://azure.microsoft.com/en-us/blog/announcing-azure-pipelines-with-unlimited-ci-cd-minutes-for-open-source/ 
I use AzurePipelines in my day by day job...if you link the repo and add me as admin of pipeline I could try to setup.

**Comment by MarcoRossignoli** [comment]
Another plus of this approach is that we can run CI on a list of SDK which could produce different IL and invalidate code assumptions and again in future we could have different instrumentation related to different SDK(we can read it runtime).

**Comment by MarcoRossignoli** [comment]
@tonerdo PTAL

**Comment by MarcoRossignoli** [comment]
@tonerdo merged and fixed azp

**Comment by MarcoRossignoli** [comment]
@tonerdo  woot command works!

**Comment by MarcoRossignoli** [comment]
@tonerdo I think it's ready to go!

**Comment by AArnott** [comment]
I wonder if you're trying too hard. Your repo's build requirements really aren't that strange to justify all these custom build scripts.

Add a global.json file to your repo to "pin" the right .NET Core SDK used to build with. [Here's an example](https://github.com/microsoft/vs-threading/blob/master/src/global.json).

Then modify your azure pipelines YAML file to install that version. [Here's an example](https://github.com/microsoft/vs-threading/blob/8c9c7a44eb4ff6284b373fc416fc1e9dbfae4cab/azure-pipelines/build.yml#L18-L23).

**Comment by MarcoRossignoli** [comment]
@AArnott the idea of download sdk is for future use...I mean instrumenting code and verify coverage depends on compiler version, different version of compilers generage different pdb and in future we would like to have a way to run tests with more than one sdk version...for instance take a look at this PR https://github.com/tonerdo/coverlet/issues/389 https://github.com/tonerdo/coverlet/pull/390 behaviour of hide sequence point is changed and our tests started to fail.

**Comment by AArnott** [comment]
[Here's what it looks like in full for your repo](https://github.com/tonerdo/coverlet/compare/master...AArnott:PinSdkVersion?expand=1)

Disclaimer: not fully tested yet.

**Comment by AArnott** [comment]
It's now tested.

**Comment by MarcoRossignoli** [comment]
@AArnott do you have better way to provide a CI with more than one SDK?
My idea as I've explained above is the possibility to have more than one SDK, so download from a list and rebuild/test for every different version and in case amend tests with assertion related to version number(i.e. line x is not covered in version x.y.x due to different pdb).

**Comment by AArnott** [comment]
Yes, I think we can handle that too.

My personal recommendation is that you *build* with exactly one SDK version (that you've pinned for consistency). Then you *test* on as many SDKs as you need your coverlet build extension to support for your target customers.

**Comment by MarcoRossignoli** [comment]
Now CI depends on custom sdk that is one of the goal of this PR.
Close for the moment we'll pickup some script idea when/if we'll decide how to approach to this particular scenario.

**Pull request closed without merging**